src/ZoneServer/Network/Send.cs

```diff
@@ -3549,7 +3568,10 @@ public static void ZC_SOLD_ITEM_DIVISION_LIST(Character character, byte type, Li
 						zpacket.PutInt(items[i].Amount);
 						zpacket.PutInt(items[i].Price);
 						zpacket.PutInt(1);
-						zpacket.PutInt(items.Count - i - 1);
+						if (itemsPositions != null)
```

Comment:
Well ZC_SOLD_ITEM_DIVISION_LIST is sent to client both when player is browsing a npc store or the storage. The game shares some packets of npc stores for storage. In this case, the npc store cannot have empty items for sale so "items.count - i - 1" for item positions makes sense, but in storage, items can have positions that are not consecutive. Therefore, I also need to send the position of the item.

Basically I want to send that exact same packet but with one different integer logic. What I could do, instead, is duplicating the code and overload it into receiving the type "List<StorageItem>" instead of "List<Item>" which would be a wrapper around 'Item' class but with a property of the position the item is in storage.

Reply:
It does seem like creating more wrapper classes around items might be worth considering. That way we could store additional information for them, similar to how we do it in the database, without polluting the `Item` class.